byterun/caml/misc.h

```diff
@@ -149,10 +149,10 @@ extern void caml_set_fields (intnat v, unsigned long, unsigned long);
 
 /* snprintf emulation for Win32 */
 
-#ifdef _WIN32
+#if defined(_WIN32) && defined(_MSC_VER) && _MSC_VER < 1900
```

Comment:
http://sourceforge.net/p/mingw-w64/mailman/message/31241434/  seems to indicate that even though mingw provides snprintf, it might simply delegate to the non-conforming version provided by MS.

@xavierleroy You introduced caml_snprintf in the C99 switch in Aug. 2014.  Do you remember if it was required for mingw as well?